test/path2d.test.js

```diff
@@ -58,6 +60,7 @@ describe("Path2D", ()=>{
 
       let clone = new Path2D()
       for (const [verb, ...pts] of p.edges){
+        // @ts-expect-error This is not possible to be done in a type-safe way in TypeScript (yet, at least)
```

Comment:
Given that this doesn't satisfy the compiler, is there any benefit to the complexity of the Path2DEdge [declaration](https://github.com/samizdatco/skia-canvas/pull/52/files#diff-cfc7807f6b174e1c3705c5c2747f56e49a889a52999dc96e4080a3af0131a534R138-R149) over something simpler like:

```ts
export type Path2DEdge = [verb:"closePath"]
    | [verb:"moveTo", x:number, y:number]
    | [verb:"lineTo", x:number, y:number]
    | [verb:"quadraticCurveTo", cpx:number, cpy:number, x:number, y:number]
    | [verb:"bezierCurveTo", cp1x:number, cp1y:number, cp2x:number, cp2y:number, x:number, y:number]
    | [verb:"conicCurveTo", cpx:number, cpy:number, x:number, y:number, weight:number];
```

or even:
```ts
type Path2DEdge = [verb: string, ...args: number[]]
```

Reply:
The first option is what the complex code compiles to actually, so it wouldn't work in here any way -- TS is not discriminant enough yet to change the argument list type depending on the hypothetical verbs.

But yeah I agree, if it's not possible to type it yet let's just do a more general one like the second one you proposed.